pkg/doc/signature/jsonld/processor.go

```diff
@@ -103,11 +116,9 @@ func (p *Processor) GetCanonicalDocument(doc map[string]interface{}, opts ...Pro
 		return nil, fmt.Errorf("failed to normalize JSON-LD document, invalid view")
 	}
 
-	if procOptions.removeInvalidRDF {
-		result, err = p.removeMatchingInvalidRDFs(result)
-		if err != nil {
-			return nil, fmt.Errorf("failed to normalize due to invalid RDF dataset: %w", err)
-		}
+	result, err = p.removeMatchingInvalidRDFs(result, procOptions)
+	if err != nil {
+		return nil, err
```

Comment:
How do we allow the higher layer package to decide how to enable such cases? 

Are we just failing rather than enabling a warning (or otherwise enabling the higher layer package to decide how to handle)?

Reply:
Higher level packages will decide how to handle these cases.

Now signer & verifier will use default "WithValidateRDF" which validates and fails when invalid RDF found

Reply:
line number 179

Reply:
Ok - can we provide two examples in the tests:

(a) "Client" who wants to print warnings about invalid RDF but allow signing/verifier to continue. The client should print the warnings not the library, of course.
(b) "Client" who wants to fail signing/verifier on invalid RDF but have an error object that indicates this is the failure (rather than invalid signature).

Reply:
@sudeshrshetty would you want to do these examples as the next PR?

Reply:
will add it in upcoming PR